Gemfile

```diff
@@ -11,7 +11,14 @@ else
   gem "rails", "~> #{rails_version}"
 end
 
-gem "rack-test"
+group :test do
+  if RUBY_VERSION == "1.9.3"
```

Comment:
Turns out neither of our approaches really "work". Mine is broken because the main `Gemfile.lock` of the project will be different depending on what ruby version you're using. The platforms suggestion only works if the gems are named differently, otherwise you get a `You cannot specify the same gem twice with different version requirements` error.

It looks like the only real solution is multiple gemfiles (one for each supported version of ruby), and updating our `.travis.yml` appropriately. See:

http://stackoverflow.com/questions/8695532/using-bundler-to-load-different-versions-of-gems-for-different-platforms
https://github.com/travis-ci/travis-ci/issues/953